torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -230,10 +230,17 @@ static Value* tryMatchArgument(
 
 c10::optional<size_t> findInputWithName(
     const std::string& name,
-    at::ArrayRef<NamedValue> kwargs) {
+    at::ArrayRef<NamedValue> kwargs,
+    bool is_aten) {
```

Comment:
Is it necessary to check `is_aten`? What if we simply allow `input` to be an alias for `self` universally? (unless there is another argument that's called input). What do you think?

Reply:
if there's a user defined function with both `input` and `self` defined something weird could happen

Reply:
```
def f(x):
            return torch.ceil(self=x)

        inp = torch.randn(10)
        self.checkScript(f, (inp, ))
```

doesn't work on master, so should be ok

Reply:
We can have operators with input as their parameter name in the future. So it is probably better to do it for only aten ops for now? @gmagogsfm

Reply:
Yeah, we should only allow this aliasing when there is no other argument with `self`, `input` as their names. That should be safe.